plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -68,155 +69,198 @@ open class DokkaDescriptorVisitor( // TODO: close this class and make it private
             dri = driWithPlatform.dri,
             functions = scope.functions(driWithPlatform),
             properties = scope.properties(driWithPlatform),
-            classlikes = scope.classlikes(driWithPlatform)
+            classlikes = scope.classlikes(driWithPlatform),
+            packages = scope.packages(driWithPlatform),
+            documentation = descriptor.resolveDescriptorData(platformData),
+            platformData = listOf(platformData)
         )
     }
 
     override fun visitClassDescriptor(descriptor: ClassDescriptor, parent: DRIWithPlatformInfo): Classlike =
         when (descriptor.kind) {
-            org.jetbrains.kotlin.descriptors.ClassKind.ENUM_CLASS -> enumDescriptor(descriptor, parent)
+            ClassKind.ENUM_CLASS -> enumDescriptor(descriptor, parent)
+            ClassKind.ENUM_ENTRY -> enumDescriptor(descriptor, parent)
+            ClassKind.OBJECT -> objectDescriptor(descriptor, parent)
+            ClassKind.INTERFACE -> interfaceDescriptor(descriptor, parent)
             else -> classDescriptor(descriptor, parent)
         }
 
+    private fun interfaceDescriptor(descriptor: ClassDescriptor, parent: DRIWithPlatformInfo): Interface {
+        val driWithPlatform = parent.dri.withClass(descriptor.name.asString()).withEmptyInfo()
+        val scope = descriptor.unsubstitutedMemberScope
+        val info = descriptor.resolveClassDescriptionData(platformData)
+
+        return Interface(
+            dri = driWithPlatform.dri,
+            name = descriptor.name.asString(),
+            functions = scope.functions(driWithPlatform),
+            properties = scope.properties(driWithPlatform),
+            classlikes = scope.classlikes(driWithPlatform),
+            actual = descriptor.actual(platformData, descriptor),
+            visibility = PlatformDependent.from(platformData, descriptor.visibility),
```

Comment:
For psi parsing I created an extension function for simplifying `PlatformDependent.from(platformData, <something>)`, you can reuse it here